lib/charms/hydra/v0/oauth.py

```diff
@@ -201,11 +202,29 @@ def _dump_data(data: Dict, schema: Optional[Dict] = None) -> Dict:
                 ret[k] = json.dumps(v)
             except json.JSONDecodeError as e:
                 raise DataValidationError(f"Failed to encode relation json: {e}")
+        elif isinstance(v, bool):
+            ret[k] = str(v)
         else:
             ret[k] = v
     return ret
 
 
+def strtobool(val: str) -> bool:
```

Comment:
`ValueError` may be raised instead of returning bool
```suggestion
def strtobool(val: str) -> Optional[bool]:
```